torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -506,6 +506,31 @@ class AttributePropagator {
     return false;
   }
 
+  void removeExtraWaitCalls(Block* b) {
+    auto nodes = b->nodes();
+    for (auto it = nodes.begin(); it != nodes.end(); it++) {
+      auto node = *it;
+      if (node->kind() != aten::wait) {
+        continue;
+      }
+      TORCH_INTERNAL_ASSERT(node->inputs().size() == 1);
```

Comment:
Ideally, freezing wouldn't be passes invalid IR, and any pass that introduces an `aten::wait` inconsistency would do it's own cleanup

Reply:
@eellison makes sense, I'll move this pass out from freezing in a follow-up PR.